pkg/controller/job/job_controller.go

```diff
@@ -486,10 +486,11 @@ func (jm *Controller) syncJob(key string) (bool, error) {
 	}
 	activePods := controller.FilterActivePods(pods)
 	active := int32(len(activePods))
+	suspended := utilfeature.DefaultFeatureGate.Enabled(features.SuspendJob) && job.Spec.Suspend
 	succeeded, failed := getStatus(&job, pods)
-	conditions := len(job.Status.Conditions)
-	// job first start
-	if job.Status.StartTime == nil {
+	// Job first start. Set StartTime and start the ActiveDeadlineSeconds timer
+	// only if the job is not in the suspended state.
+	if job.Status.StartTime == nil && !suspended {
```

Comment:
discussed this on slack, we've agreed to reset StartTime every time a Job is resumed because: (1) it keeps the code simpler, (2) resuming a job created in the suspended state sets StartTime only when it's resumed, so this would offer a similar behaviour